TimberAPI/AssetLoaderSystem/ResourceAssetPatch/TimberApiResourceAssetLoader.cs

```diff
@@ -11,15 +10,30 @@ public class TimberApiResourceAssetLoader : IResourceAssetLoader
     {
         private readonly IAssetLoader _assetLoader;
 
-        private Shader _shader;
+        private static Shader _buidlingShader;
+        private static Shader _dirtShader;
+        private static Shader _pathShader;
 
-        private Shader Shader
+        private static Shader Shader(Material mat)
         {
-            get
+           
+            if (_buidlingShader == null)
+                _buidlingShader = Resources.Load<GameObject>("Buildings/Paths/Platform/Platform.Full.Folktails").GetComponent<MeshRenderer>().materials[0].shader;
+            if (_dirtShader == null)
+                _dirtShader = Resources.Load<GameObject>("Buildings/Wellbeing/Carousel/CarouselConstructionSite5x5+3x1Dirt").GetComponentsInChildren<MeshRenderer>()[1].materials[0].shader;
+            if (_pathShader == null)
+                _pathShader = Resources.Load<GameObject>("Buildings/Paths/Path/DirtDrivewayStraightPath").GetComponent<MeshRenderer>().materials[0].shader;
+            
+            switch (mat.renderQueue)
             {
-                if (_shader == null)
-                    _shader = Resources.Load<GameObject>("Buildings/Paths/Platform/Platform.Full.Folktails").GetComponent<MeshRenderer>().materials[0].shader;
-                return _shader;
+                case 2999:
```

Comment:
While maybe barely excused in this case, magic numbers should be avoided. Maybe using an enum here instead is better? Something like:
```
public enum RenderQueueType
{
    Dirt = 2999,
    Path = 2998
}
...
switch (mat.renderQueue)
{
    case RenderQueueType.Dirt:
       ...
}
```

Reply:
This does make it more readable. I changed it :D